File: 896. Monotonic Array/896.go
package main

import (
	"fmt"
)

func main(){
	fmt.Println(isMonotonic([]int{3,3,3}))
}

func isMonotonic(A []int) bool {
	flag := 0
	pre := A[0]
	for _,v := range A{
		if v>pre{
			if flag < 0{
				return false
			}
			flag = 1
		}
		if v < pre{
			if flag > 0{
				return false
			}
			flag = -1
		}
		pre = v
	}
	return true
}

File: 896. Monotonic Array/896.py
class Solution:
    def isMonotonic(self, A) -> bool:
        flag = 0
        pre = A[0]
        for i in A:
            if i>pre:
                if flag == -1:
                    return False
                pre = i
                flag = 1
            elif  i<pre:
                if flag == 1:
                    return False
                pre = i
                flag = -1
            else:
                pre = i
        return True
    

a = Solution()
print(a.isMonotonic([1,1,1]))